src/hooks/useLocalStorage.ts

```diff
@@ -9,23 +11,39 @@ function useLocalStorage<T>(
   key: string
 ): [T | null, Dispatch<SetStateAction<T | null>>];
 function useLocalStorage<T>(key: string, defaultValue?: T) {
-  const [value, setValue] = useState(() => {
-    if (typeof window === "undefined") {
-      return defaultValue ?? null;
-    }
-    const storedValue = getLocalStorageItem<T>(key);
-
-    if (!storedValue) return defaultValue ?? null;
-    return storedValue;
+  const [storedValue, setStoredValue] = useState<T | null>(() => {
+    return defaultValue ?? null;
   });
 
   useEffect(() => {
-    if (typeof window !== "undefined") {
-      setLocalStorageItem(key, value);
+    if (!isServer) {
```

Comment:
```suggestion
    if (isServer === true) {
      return;
    }
```
요렇게 Early Return을 사용해주시면 Block Depth도 더 얕아지고, 코드를 읽으며 예외에 대한 정보를 순차적으로 처리할 수 있어서 가독성에도 좋을 것 같습니다!